LibRangeCheck-2.0/LibRangeCheck-2.0.lua

```diff
@@ -1030,29 +1047,46 @@ function lib:GetRange(unit, checkVisible, noItems)
         return nil
     end
 
+    local currentTime = GetTime()
+    local guid = UnitGUID(unit)
+    local cacheItem = rangeCache[guid]
+    if cacheItem and cacheItem.updateTime + self.getRangeThrottle > currentTime then
+        return cacheItem.minRange, cacheItem.maxRange
+    end
+
+    local result = cacheItem or {}
+
     local canAssist = UnitCanAssist("player", unit)
     if UnitIsDeadOrGhost(unit) then
         if canAssist then
-            return getRange(unit, self.resRC)
+            result.minRange, result.maxRange = getRange(unit, self.resRC)
         else
-            return getRange(unit, self.miscRC)
+            result.minRange, result.maxRange = getRange(unit, self.miscRC)
         end
     end
 
     if UnitCanAttack("player", unit) then
-        return getRange(unit, noItems and self.harmNoItemsRC or self.harmRC)
+        result.minRange, result.maxRange = getRange(unit, noItems and self.harmNoItemsRC or self.harmRC)
     elseif UnitIsUnit("pet", unit) then
-        local minRange, maxRange = getRange(unit, noItems and self.friendNoItemsRC or self.friendRC)
-        if minRange or maxRange then
-            return minRange, maxRange
-        else
-            return getRange(unit, self.petRC)
+        result.minRange, result.maxRange = getRange(unit, noItems and self.friendNoItemsRC or self.friendRC)
+        if not result.minRange and not result.maxRange then
+            result.minRange, result.maxRange = getRange(unit, self.petRC)
         end
     elseif canAssist then
-        return getRange(unit, noItems and self.friendNoItemsRC or self.friendRC)
+        result.minRange, result.maxRange = getRange(unit, noItems and self.friendNoItemsRC or self.friendRC)
     else
-        return getRange(unit, self.miscRC)
+        result.minRange, result.maxRange = getRange(unit, self.miscRC)
     end
+
+    result.updateTime = currentTime
+    rangeCache[guid] = result
+    return result.minRange, result.maxRange
```

Comment:
By returning the cached values so early some checks are lost
The cache logic could be refactored in it's own function between lib.GetRange and getRange

Reply:
I don't really see which checks are relevant to do again if we cache by GUID. Because the checks only determine how we do the range check. Which we try to avoid with caching anyway.

Reply:
Ah I see it now, the `noItems` argument is ignored. I think it would make sense to add this argument to the cache key.